op-node/rollup/derive/calldata_source.go

```diff
@@ -129,10 +131,20 @@ func DataFromEVMTransactions(config *rollup.Config, daCfg *rollup.DAConfig, batc
 				log.Warn("unable to decode data pointer", "index", j, "err", err)
 				continue
 			}
-			data, err := daCfg.Client.NamespacedData(context.Background(), daCfg.NamespaceId, uint64(height))
+
+			var data [][]byte
```

Comment:
As per this function docs:

`// This will return an empty array if no valid transactions are found.`

So maybe we can avoid the retry loop by just returning a empty array instead.

Reply:
if it fails to fetch the data, then it will need to fetch the data, so I think retry is necessary?

Reply:
Ideally we should handle this more robustly, as there is already a pattern optimism uses for connection errors by passing `TemporaryError`:

https://gist.github.com/tuxcanfly/6d2a298d975f992831ca830bea1375d3

But we can follow up on that in a another PR.

Reply:
we can do it in this one, i just need help if we do